File: drills/ts/harnesses/nc97_interleaving_string.ts
import * as solution from "./solution";

const show = (v: unknown) => JSON.stringify(v) ?? "undefined";

export function run(): [string, string, string][] {
  if (typeof solution.isInterleave !== "function") throw new Error("__signature_mismatch__");
  return [
    ["isInterleave('aabcc', 'dbbca', 'aadbbcbcac')", show(true), show(solution.isInterleave('aabcc', 'dbbca', 'aadbbcbcac'))],
    ["isInterleave('aabcc', 'dbbca', 'aadbbbaccc')", show(false), show(solution.isInterleave('aabcc', 'dbbca', 'aadbbbaccc'))],
    ["isInterleave('', '', '')", show(true), show(solution.isInterleave('', '', ''))],
    ["isInterleave('a', '', 'a')", show(true), show(solution.isInterleave('a', '', 'a'))],
    ["isInterleave('', 'b', 'b')", show(true), show(solution.isInterleave('', 'b', 'b'))],
    ["isInterleave('abc', 'def', 'adbecf')", show(true), show(solution.isInterleave('abc', 'def', 'adbecf'))],
  ];
}

File: drills/go/harnesses/nc97_interleaving_string.go
package main

func main() {
	run(func() []testCase {
		return []testCase{
			tc("isInterleave('aabcc', 'dbbca', 'aadbbcbcac')", true, isInterleave("aabcc", "dbbca", "aadbbcbcac")),
			tc("isInterleave('aabcc', 'dbbca', 'aadbbbaccc')", false, isInterleave("aabcc", "dbbca", "aadbbbaccc")),
			tc("isInterleave('', '', '')", true, isInterleave("", "", "")),
			tc("isInterleave('a', '', 'a')", true, isInterleave("a", "", "a")),
			tc("isInterleave('', 'b', 'b')", true, isInterleave("", "b", "b")),
			tc("isInterleave('abc', 'def', 'adbecf')", true, isInterleave("abc", "def", "adbecf")),
		}
	})
}
